src/category_theory/triangulated/pretriangulated.lean

```diff
@@ -0,0 +1,204 @@
+/--
+Given any distinguished triangle `T`, then we know `T.rotate` is also distinguished.
+-/
+lemma rot_of_dist_triangle (T ∈ dist_triang C) : (T.rotate ∈ dist_triang C) :=
+(rotate_distinguished_triangle T).mp H
+
+/--
+Given any distinguished triangle `T`, then we know `T.inv_rotate` is also distinguished.
+-/
+lemma inv_rot_of_dist_triangle (T ∈ dist_triang C) : (T.inv_rotate ∈ dist_triang C) :=
+(rotate_distinguished_triangle (T.inv_rotate)).mpr
+  (isomorphic_distinguished T H (T.inv_rotate.rotate) T (inv_rot_comp_rot.symm.app T))
+
+/--
+Given any distinguished triangle
+```
+      f       g       h
+  X  ───> Y  ───> Z  ───> X⟦1⟧
+```
+the composition `f ≫ g = 0`.
+See https://stacks.math.columbia.edu/tag/0146
+-/
+lemma comp_dist_triangle_mor_zero₁₂ (T ∈ dist_triang C) : T.mor₁ ≫ T.mor₂ = 0 :=
+begin
+  have h := contractible_distinguished T.obj₁,
+  have f := complete_distinguished_triangle_morphism,
+  specialize f (contractible_triangle C T.obj₁) T h H (𝟙 T.obj₁) T.mor₁,
+  have t : (contractible_triangle C T.obj₁).mor₁ ≫ T.mor₁ = 𝟙 T.obj₁ ≫ T.mor₁,
+    by refl,
+  specialize f t,
+  cases f with c f,
+  rw ← f.left,
+  simp only [limits.zero_comp, contractible_triangle_mor₂],
+end -- TODO : tidy this proof up
+
+/--
+Given any distinguished triangle
+```
+      f       g       h
+  X  ───> Y  ───> Z  ───> X⟦1⟧
+```
+the composition `g ≫ h = 0`.
+See https://stacks.math.columbia.edu/tag/0146
+-/
+lemma comp_dist_triangle_mor_zero₂₃  (T ∈ dist_triang C) : T.mor₂ ≫ T.mor₃ = 0 :=
+comp_dist_triangle_mor_zero₁₂ C T.rotate (rot_of_dist_triangle C T H)
+
+/--
+Given any distinguished triangle
+```
+      f       g       h
+  X  ───> Y  ───> Z  ───> X⟦1⟧
+```
+the composition `h ≫ f⟦1⟧ = 0`.
+See https://stacks.math.columbia.edu/tag/0146
+-/
+lemma comp_dist_triangle_mor_zero₃₁ (T ∈ dist_triang C) :
+  T.mor₃ ≫ ((shift C).functor.map T.mor₁) = 0 :=
+have H₂ : _ := rot_of_dist_triangle C T.rotate (rot_of_dist_triangle C T H),
+by simpa using comp_dist_triangle_mor_zero₁₂ C (T.rotate.rotate) H₂
+
+/-
+TODO: If `C` is pretriangulated with respect to a shift,
+then `Cᵒᵖ` is pretriangulated with respect to the inverse shift.
+-/
+end pretriangulated
+end category_theory.triangulated
+
+namespace category_theory.triangulated
+namespace pretriangulated
+
+variables (C : Type u₁) [category.{v₁} C] [has_shift C] [additive_category C]
+[functor.additive (shift C).functor] [functor.additive (shift C).inverse]
+variables (D : Type u₂) [category.{v₂} D] [has_shift D] [additive_category D]
+[functor.additive (shift D).functor] [functor.additive (shift D).inverse]
+
+/--
+A triangulated functor between pretriangulated categories `C` and `D` is a functor `F : C ⥤ D`
+together with given functorial isomorphisms `ξ X : F(X⟦1⟧) ⟶ F(X)⟦1⟧` with extra conditions
+involving images of triangles.
+-/
+structure triangulated_functor_struct extends (C ⥤ D) :=
+(nat_iso : (shift C).functor ⋙ to_functor ≅ to_functor ⋙ (shift D).functor)
```

Comment:
This name could still be better. How about `comm_shift`? (Or `shift_comm`, or `comm_shift_iso`?)